craft_parts/plugins/properties.py

```diff
@@ -65,4 +81,7 @@ def get_pull_properties(cls) -> list[str]:
     def get_build_properties(cls) -> list[str]:
         """Obtain the list of properties affecting the build stage."""
         properties = cls.schema(by_alias=True).get("properties")
-        return list(properties.keys()) if properties else []
+        if properties:
+            del properties["plugin"]
```

Comment:
how safe is this?

Reply:
It should be safe (the `plugin` key not existing in the properties dict would mean we have a much bigger problem), but I rewrite it with a list comprehension anyway since that's faster and safer

Reply:
(And if you meant because of the `del`, yes that's safe too, as `schema` creates a new dict each time.)